cli/azd/internal/agent/consent/checker.go

```diff
@@ -0,0 +1,452 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package consent
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/ux"
+	"github.com/mark3labs/mcp-go/mcp"
+)
+
+// ConsentChecker provides shared consent checking logic for different tool types
+type ConsentChecker struct {
+	consentMgr ConsentManager
+	serverName string
+}
+
+// NewConsentChecker creates a new shared consent checker
+func NewConsentChecker(
+	consentMgr ConsentManager,
+	serverName string,
```

Comment:
IMO we don't this `ConsentManager`. It is only helping to get `serverName` in the state for all methods.
I would merge this Checker with the `ConsentManager`, which defines the argument-options like `ConsentRequest`.